style.md

```diff
@@ -1042,6 +1043,136 @@ func TestSigner(t *testing.T) {
+implementations.
+Instead, hand-write only the methods your lists will delegate to the abstract
+list.
+
+```go
+type AbstractList struct {}
+
+// Add adds an entity to the list.
+func (l *AbstractList) Add(e Entity) {
+    // ...
+}
+
+// Remove removes an entity from the list.
+func (l *AbstractList) Remove(e Entity) {
+    // ...
+}
+```
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+// ConceteList is a list of entities.
+type ConceteList struct {
+    *abstractlist.AbstractList
+}
+```
+
+</td><td>
+
+```go
+// ConceteList is a list of entities.
+type ConceteList struct {
+    list *abstractlist.AbstractList
+}
+
+// Add adds an entity to the list.
+func (l *ConceteList) Add(e Entity) {
+    return l.list.Add(e)
+}
+
+// Remove removes an entity from the list.
+func (l *ConceteList) Remove(e Entity) {
+    return l.list.Remove(e)
+}
+```
+
+</td></tr>
+</tbody></table>
+
+To provide a middle ground between inheritance and composition, Go allows for
+types to be [embedded] in structs.
+The type gains the methods of the embedded type.
+These methods delegate to the composite type.
+
+  [embedded]: https://golang.org/doc/effective_go.html#embedding
+
+The struct also gains a field by the same name as the type.
+Consequently, if the embedded type is public, the field is public.
+The embedded type becomes a constraint on the evolution of the container's
+public interface.
+
+An embedded type is rarely necessary.
+It is a convenience that implies monotonous delegate methods.
+
+Every future version of `ConceteList` is obliged indefinitely to embed `AbstractList`,
+eliminating the possibility of replacing the implementation with an alternative
+in a future version.
+
+Embedding an AbstractList *interface*, instead of the struct directly, would
+offer the developer more flexibility to change in the future.
+The embedded interface also leaks the implementation detail that the list uses
+an abstract list at all, which is of no concern to the end user and could
+otherwise change in a future version.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+// AbstractList is a generalized implementation
+// for various kinds of lists of entities.
+type AbstractList interface {
+    Add(Entity)
+    Remove(Entity)
+}
+
+// ConceteList is a list of entities.
+type ConceteList struct {
+    AbstractList
+}
+```
+
+</td><td>
+
+```go
+// ConceteList is a list of entities.
+type ConceteList struct {
+    list *AbstractList
+}
+
+// Add adds an entity to the list.
+func (l *ConceteList) Add(e Entity) {
+    return l.list.Add(e)
+}
+
+// Remove removes an entity from the list.
+func (l *ConceteList) Remove(e Entity) {
+    return l.list.Remove(e)
+}
+```
+
+</td></tr>
+</tbody></table>
+
+Although writing these delegate methods is tedious, it leaves more
+opportunities for change open and also eliminates indirection for discovering
```

Comment:
nit: I'd probably mention that this avoids leaking implementation details again.